### ISSUE TITLE
Twisted callback error on first launch of Consular

### ISSUE THREAD

**Issue by JayH5** [universalcore/consular#19]
The `callbackUrls` field seems to be missing on the first request to Marathon at `/v2/eventSubscriptions` for some reason...

```
2015-08-07 10:49:41+0200 [-] Unhandled error in Deferred:
2015-08-07 10:49:41+0200 [-] Unhandled Error
    Traceback (most recent call last):
      File "/var/praekelt/consular-python/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 393, in callback
        self._startRunCallbacks(result)
      File "/var/praekelt/consular-python/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 501, in _startRunCallbacks
        self._runCallbacks()
      File "/var/praekelt/consular-python/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 588, in _runCallbacks
        current.result = callback(current.result, *args, **kw)
      File "/var/praekelt/consular-python/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1184, in gotResult
        _inlineCallbacks(r, g, deferred)
    --- <exception caught here> ---
      File "/var/praekelt/consular-python/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1126, in _inlineCallbacks
        result = result.throwExceptionIntoGenerator(g)
      File "/var/praekelt/consular-python/local/lib/python2.7/site-packages/twisted/python/failure.py", line 389, in throwExceptionIntoGenerator
        return g.throw(self.type, self.value, self.tb)
      File "/var/praekelt/consular/consular/main.py", line 75, in register_marathon_event_callback
        existing_callbacks = yield self.get_marathon_event_callbacks()
      File "/var/praekelt/consular-python/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 588, in _runCallbacks
        current.result = callback(current.result, *args, **kw)
      File "/var/praekelt/consular/consular/main.py", line 61, in <lambda>
        d.addCallback(lambda data: data['callbackUrls'])
    exceptions.KeyError: 'callbackUrls'
```

**Comment by smn** [comment]
Have you configured marathon to enable callbacks with `--event_subscriber http_callback` either on the command line or in the relevant `/etc/marathon` directory?

**Comment by JayH5** [comment]
After configuring the `event_subscriber` I still get the same error.
Probably should have pointed out before: Consular still keeps running (and works) after this error.

**Comment by smn** [comment]
@JayH5 can you post the output of the [eventSubscriptions](https://mesosphere.github.io/marathon/docs/rest-api.html#get-v2-eventsubscriptions) API?

**Comment by JayH5** [comment]
@smn ah my bad, had the config at the wrong path. No error now.

**Comment by smn** [comment]
:+1: I'm still going to re-open this as the stack trace is unhelpful and we could easily print out an more sane error describing the problem.

**Comment by smn** [comment]
Contributions welcome ;)

**Comment by JayH5** [comment]
Should be solved by #20
